### PR TITLE
Dwarf Fort "Steelrest", iron wrist/hand/foot plate armor, role support for fort and couple of steel armor item renames

### PR DESCRIPTION
**Pull request by DerFlammenwerfer**
## About The Pull Request
First off, this adds iron variants of wrist, hand, and foot armor. It also renames plate chausses to steel cuisses, and plated boots to steel sabatons.

The dwarf fortress known as Steelrest is now embedded in the rock face off the northern shore of the cave lake, just east of the bog tunnel. It reaches from the cave lake all the way up to the mountain-top, but is designed to be initially compact, with space enough for 8-9 dwarves. It has the following roles/amenities:

### Roles
1. **Expedition Leader** (1 slot Dwarf; Middle-Aged or Old) - A well-educated and aging dwarf who has been around since Steelrest's founding, and may well have had a say in its architecture. Possesses mastery in literature, carpentry, masonry, and engineering, and is an expert with an axe. De-facto leader of the fort, and a yeoman of the mountain. Possesses a single riddle of steel.
2. **Provisioner** (1 slot, Dwarf, Adult or older) - A literate craftsdwarf who is very skilled with a needle, and in general craftsdwarfship besides. Also a competent cook and novice smith, responsible for ensuring that Steelrest's dwarves stay supplied and in good repair. Incidentally, also the second-in-command of the fortress.
3. **Emissary** (1 slot, Dwarf or Humen, Adult or older) - The King's liason to the fortress of Steelrest, which used to keep the bannermen well-supplied with weapons and armor before its great foundry collapsed into the volcano. Responsible for trying to maintain peaceful and positive relations and reconciling the interests of both the Crown and the dwarves.
4. **Guardians** (4 slots, Dwarf, Adult or Middle-Aged) - An iron-clad squad of militants consisting of 2 Bowdwarves and 2 Guardsdwarves. Bowdwarves have recurves and quivers; Guardsdwarves possess kite shields and steel flails. Responsible for protecting Steelrest and its dwarves.
5. **Tunnelers** (2 slots, Dwarf, Adult or older.) - A pair of survivors from the collapse of Malum's Terrace, who are very athletic and strong. Chiefly responsible for collecting and refining ore.

### Amenities

- Kitchen with 2 ovens, 2 hearths, frying pan, fishing rod, spade, skewer, cleaver, tables, chest, and millstone.
- Smeltery with 4 furnaces.
- Offices for Emissary and Expedition Leader; Emissary has access to SCOM
- Infirmary with 4 shitty beds, needle of pestra, water barrel, and some cloth. Also a loom and shelving in its closet.
- Communal quarters with 4 nice beds and a chest for each as a footlocker.
- Food storehouse with 2 cat (LOL) to keep vermin out. Also contains 1 set of farm tools and some seeds.
- Access to clean water and limited farm space inside the fort itself
- Access to mountaintop for bigger farming/ranching operations, with a bit of work
- Trade gate on north side of the fort, facing Rockhill. Access to caves from south gate, as well as small trading window there.
- Buttresses overlooking 2 of the 3 entrances, murderholes included.
- Sluiceway, which doubles as 1-way evacuation route to Rockhill.
- 3rd floor largely undeveloped, with lots of room for expansion if fort population grows

### What it specifically lacks
- Space for more than 9 people to live; needs to be expanded with the population
- Smithy to replace broken or lost equipment
- Stockpiled food, wood, fiber, metal, or alcohol
- Defenses around mountaintop, other than the topmost exit

## Why It's Good For The Game
I could go on and on about the merits of people wanting to be fort dwarves not having to spend 2/3s of the round just laying down infrastructure, but I've done that already in the discord and I'm not doing it again here. I cannot promise that this will not have the goblin problem of drawing everyone's attention, but I can at least promise that the attention will ultimately still be close to Rockhill and the current Bog, instead of all the way on the other end of the map. I also expect that some dwarf mains may still wish to build their own forts in other parts of the map, and that's fine too, but this fort is designed with the hope that if players want to change something about the layout or expand upon it, they can do so. Since this fort has roles that can roundstart or latejoin within it, the hope is that people wanting to RP as migrants from other mountainhomes will come join in.

### CODE REVIEW

**Comment by DerFlammenwerfer** [comment]
Here are some screenshots of the fort itself.

image
image
image
image

**Comment by wgtjunior743** [comment]
Very nice!

**Review by wgtjunior743 — COMMENTED** [review]
Just fix these and the PR won't run into a build failure

File: code/modules/jobs/job_types/roguetown/dwarffort/dwarves.dm [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+		H.mind.adjust_skillrank(/datum/skill/misc/reading, 6, TRUE)// Like a steward, but skilled in construction and with axes/maces instead of knives.
+		H.mind.adjust_skillrank(/datum/skill/misc/riding, 2, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/combat/crossbows, 2, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/combat/wrestling, 2, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/combat/unarmed, 2, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/misc/swimming, 1, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/misc/climbing, 1, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/misc/athletics, 1, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/combat/swords, 1, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/combat/axesmaces, 4, TRUE)
```
[the remark this answers is not in the document]
Comment (wgtjunior743) [reply code/modules/jobs/job_types/roguetown/dwarffort/dwarves.dm:48]: there are more like 2 with that

File: code/modules/jobs/job_types/roguetown/dwarffort/dwarves.dm [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+		H.mind.adjust_skillrank(/datum/skill/misc/swimming, 1, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/misc/climbing, 1, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/misc/athletics, 1, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/combat/swords, 1, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/craft/cooking, 1, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/craft/crafting, 2, TRUE)
+		H.change_stat("intelligence", 2)
+		H.change_stat("perception", 2)
+		H.change_stat("speed", -1)
+	ADD_TRAIT(H, RTRAIT_NOBLE, TRAIT_GENERIC)
```
[the remark this answers is not in the document]
Comment (wgtjunior743) [reply code/modules/jobs/job_types/roguetown/dwarffort/dwarves.dm:304]: and a bunch of RTRAIT

**Comment by Takyon69** [comment]
Gotta post those iron sprites too so I can take a look at them.

**Comment by Xkallubar** [comment]
Hell yeah. Strike the earth!

**Comment by ophaq** [comment]
This is great, but if this gets merged I am going to miss the voting of leaders and such. Also we are going to need a Malum worshiping cleric role who represents the church.

**Comment by bobjension** [comment]
mfers DID NOT learn their lessons from job-selectable goblins

File: code/modules/jobs/job_types/roguetown/dwarffort/dwarves.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (DerFlammenwerfer) [reply code/modules/jobs/job_types/roguetown/dwarffort/dwarves.dm]: Should be fixed now

File: code/modules/jobs/job_types/roguetown/dwarffort/dwarves.dm (hunk 2 above)
[the remark this answers is not in the document]
Comment (DerFlammenwerfer) [reply code/modules/jobs/job_types/roguetown/dwarffort/dwarves.dm]: Should be fixed now.

**Comment by DerFlammenwerfer** [comment]
> Gotta post those iron sprites too so I can take a look at them.
Codersprites, but I did try to stick with the palate and lighting. The on-mobs currently use the steel versions as a placeholder because there are so many onmobs variants

image
image
image

**Comment by DerFlammenwerfer** [comment]
> Also as cool as this PR is, I have a feeling that a lot of population is just going to head here to dwarf fort instead of contributing to the main town, making it more empty. Definitely worth trying for a while though, I really want to play this.
This _might_ happen for a few days just because it is shiny new content, but since the dwarves actually live pretty close to town instead of on the opposite side of the map, I don't think there will necessarily be a lack of contribution to the town.

**Comment by wgtjunior743** [comment]
> mfers DID NOT learn their lessons from job-selectable goblins

different from goblins a decent part of dwarves try to make a dwarf fortress most of the rounds, so it is just going to take these players and give them an official role, that is it, now the goblins were a totally different story

**Comment by ophaq** [comment]
Someone please speedmerge this I must have it.

**Comment by CVSthePharmacy** [comment]
give the expedition leader the grudge book

**Comment by ChungusGamer666** [comment]
Enough with the role slop. Giving people free riddle of steel is outrageously retarded and this does not have good additions to warrant being a thing.

**Comment by ChungusGamer666** [comment]
Stop splitting the fucking playerbase. This is literally just goblins v2.


## I don't want a second fucking kingdom especially now that the player count is going down!

**Comment by BlueMemesauce** [comment]
why would a human kingdom have a random dwarf fort that's bigger than the fucking kings manor

**Comment by ophaq** [comment]
Damn this PR was so cool how could you do this to us.

**Pull request closed without merging**